core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -51,6 +54,15 @@
 
     private static final Logger LOGGER = Logger.getLogger(AtomicFileWriter.class.getName());
 
+    private static /* final */ boolean LOSE_INTEGRITY = SystemProperties.getBoolean(
+            AtomicFileWriter.class.getName() + ".LOSE_DATA_INTEGRITY_FOR_PERFORMANCE");
```

Comment:
This name feels really overly dramatic.   Could we call it "DISABLE_FORCED_FLUSH" instead?

Reply:
Basic rule: an option should tell people something about what it does -- this isn't a very descriptive name IMO and isn't really accurate either since it's not globally destroying data integrity, it's modifying one specific behavior wrt the OS caching.

Reply:
I agree to adjust this, but I think our current behaviour is also not 100% safe enough by default, which it should probably be. I could have put something like `REDUCE_LIKELINESS_TO_LOSE_DATA_WHEN_THINGS_GO_WRONG` and that would indeed have been closer to the reality, but well :-).

`DISABLE_FORCED_FLUSH` feels a bit not dramatic enough. My thought and understanding of this is that we really would like to not have it disabled apart from emergencies and very special cases, so let's not encourage it.

But OK, let's rename the option, and still leave a (less dramatic too, ok) message on class loading.